management/src/main/scala/akka/management/AkkaManagement.scala

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.management
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.Done
+import akka.actor.{ ActorSystem, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.event.Logging
+import akka.http.javadsl.HttpsConnectionContext
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.server.Directives.{ authenticateBasicAsync, pathPrefix, rawPathPrefix, AsyncAuthenticator }
+import akka.http.scaladsl.server.{ Directive, Directives, Route, RouteResult }
+import akka.management.http.{ ManagementRouteProvider, ManagementRouteProviderSettings }
+import akka.stream.ActorMaterializer
+
+import scala.collection.immutable
+import scala.concurrent.{ Future, Promise }
+import scala.util.{ Failure, Success, Try }
+
+object AkkaManagement extends ExtensionId[AkkaManagement] with ExtensionIdProvider {
+  override def lookup: AkkaManagement.type = AkkaManagement
+
+  override def get(system: ActorSystem): AkkaManagement = super.get(system)
+
+  override def createExtension(system: ExtendedActorSystem): AkkaManagement =
+    new AkkaManagement()(system)
+}
+
+final class AkkaManagement(implicit system: ExtendedActorSystem) extends Extension {
+  val log = Logging(system, getClass)
+  val settings = new AkkaManagementSettings(system.settings.config)
+
+  private implicit val materializer = ActorMaterializer()
+  import system.dispatcher
+
+  private val routeProviders: immutable.Seq[ManagementRouteProvider] = loadRouteProviders()
+
+  /**
+   * Set async authenticator to be used for management routes.
+   *
+   * Must be called BEFORE `start()` to take effect.
+   * Do not call concurrently.
+   */
+  def setAsyncAuthenticator(auth: AsyncAuthenticator[String]): Unit =
```

Comment:
Maybe we should even throw for the case where it's called late, so users won't be running unauthenticated without knowing it?

Reply:
Good point!